test/rubygems/test_gem_indexer.rb

```diff
@@ -31,15 +31,26 @@ def setup
     @indexer = Gem::Indexer.new(@indexerdir)
   end
 
+  def teardown
+    FileUtils.rm_rf(@indexer.directory)
+  ensure
+    super
+  end
+
   def test_initialize
     assert_equal @indexerdir, @indexer.dest_directory
-    assert_match %r{#{Dir.mktmpdir('gem_generate_index').match(/.*-/)}}, @indexer.directory
+    Dir.mktmpdir('gem_generate_index') do |tmpdir|
+      assert_match %r{#{tmpdir.match(/.*-/)}}, @indexer.directory
+    end
 
-    indexer = Gem::Indexer.new @indexerdir
-    assert indexer.build_modern
+    indexer1 = Gem::Indexer.new @indexerdir
+    assert_predicate indexer1, :build_modern
 
-    indexer = Gem::Indexer.new @indexerdir, :build_modern => true
-    assert indexer.build_modern
+    indexer2 = Gem::Indexer.new @indexerdir, :build_modern => true
+    assert_predicate indexer2, :build_modern
+  ensure
+    FileUtils.rm_rf(indexer1.directory)
+    FileUtils.rm_rf(indexer2.directory)
```

Comment:
Samething about making sure `indexer1` and `indexer2` are defined when this `ensure` block is run.